hibernate-rx-core/src/main/java/org/hibernate/rx/impl/PoolConnection.java

```diff
@@ -58,25 +58,25 @@ public PoolConnection(Pool pool, boolean showSQL) {
 	}
 
 	@Override
-	public CompletionStage<Optional<Integer>> updateReturning(String sql, Tuple parameters) {
+	public CompletionStage<Optional<Long>> updateReturning(String sql, Tuple parameters) {
 		return preparedQuery( sql, parameters )
-				.thenApply( rows -> Optional.ofNullable( rows.property(MySQLClient.LAST_INSERTED_ID) ) );
+				.thenApply( rows -> Optional.ofNullable( rows.property( MySQLClient.LAST_INSERTED_ID ) ) );
 	}
 
 	@Override
 	public CompletionStage<RowSet<Row>> preparedQuery(String sql, Tuple parameters) {
 		if (showSQL) {
 			System.out.println(sql);
 		}
-		return pool.preparedQuery( sql, parameters );
+		return pool.preparedQuery( sql, parameters ).convert().toCompletionStage();
 	}
 
 	@Override
 	public CompletionStage<RowSet<Row>> preparedQuery(String sql) {
 		if (showSQL) {
 			System.out.println(sql);
 		}
-		return pool.preparedQuery( sql );
+		return pool.preparedQuery( sql ).convert().toCompletionStage();
```

Comment:
What's the point of this change if we're just going to immediately convert everything to `CompletionStage`?